### ISSUE TITLE
Fix keyboard navigation for modals 

### ISSUE THREAD

**Issue by julianguyen** [ifmeorg/ifme#2118]
# Description

When using keyboard navigation to `TAB` through a page, modals should not open up unless the `ENTER` key is hit. Currently, this is not happening when a user hits `TAB` on a button that triggers a modal, the modal opens up.

Fixing this issue should be done in the [React component for Modal](https://github.com/ifmeorg/ifme/blob/main/client/app/components/Modal/index.jsx).

# Reproduction Steps

## Current Behaviour

1. `TAB` through the entire page until you hit a button that opens a modal
2. After you are focused on the button, hit `TAB` again
3. The modal will open

## Expected Behaviour

1. `TAB` through the entire page until you hit a button that opens a modal
2. After you are focused on the button, hit `TAB` again
3. The focus should be on the next link or button on the page
4. Refresh the page and TAB` through the entire page until you hit a button that opens a modal
5. Hit `ENTER` and the modal should open up

## Contributor Blurb Modals

These appear on the non-signed in Homepage and the About page (non-signed in and signed in).

![Sep-30-2022 22-26-53](https://user-images.githubusercontent.com/3010728/193379553-7376d60c-e1cf-4d5d-bb74-8dde7644d6d1.gif)

## Notification Modal in the Dashboard Navigation

This appears on every signed-in page.

![Sep-30-2022 22-23-55](https://user-images.githubusercontent.com/3010728/193379552-06067be9-3653-40c5-9365-5dc395f5b2e3.gif)

---

Please assign yourself (via the Assignees dropdown), if you do want to work on this issue. Can't find yourself? You need to [join our organization](https://github.com/ifmeorg/ifme/wiki/Join-Our-Slack).

Check out our [Picking Up Issues](https://github.com/ifmeorg/ifme/wiki/Picking-Up-Issues) guide if you haven't already!
